src/wasm.h

```diff
@@ -1423,22 +1422,25 @@ class Memory : public Importable {
   bool exists = false;
   Name name;
   Address initial = 0; // sizes are in pages
-  Address max = kMaxSize;
+  Address max = kMaxSize32;
   std::vector<Segment> segments;
 
   // See comment in Table.
   bool shared = false;
+  bool is64 = false;
```

Comment:
Will we have `is128` eventually? :stuck_out_tongue_winking_eye: 

Jokes aside, maybe this could just be `WasmType ptrType` which is the type of indexes to the memory, and is either i32 or i64? (verified by the validator)

Reply:
I prefer enums also, but this really is never going to get any additional values. `is64` is how it is represented in LLVM, WABT also, much like `shared` is a bool derived from the limits.

Reply:
I made it store the type, and the bool being a utility function